spec/bitcoin.did

```diff
@@ -0,0 +1,61 @@
+type Satoshi = nat64;
+
+type OutPoint = record {
+  txid : blob;
+  vout : nat32
+};
+
+type Utxo = record {
+  outpoint: OutPoint;
+  value: Satoshi;
+  height: nat32;
+  confirmations: nat32;
+};
+
+type GetUtxosRequest = record {
+  address : text;
+  min_confirmations: opt nat32;
+  offset: opt nat32;
+};
+
+
+type GetUtxosError = variant {
+  MalformedAddress;
+  // More error types to be added here.
```

Comment:
We cannot evolve this in a backward compatible way by _extending_ the variant we return, to the best of my knowledge. @crusso and @chenyan-dfinity should be the experts here. I think we have to provide the exhaustive list from the beginning.

Reply:
Yes, I agree.

It's possible that having wrapped each Err in opt will allow you to add a variant  in those options in a future version of Candid, but I believe that in the current version, all old clients wll received `null` if a new implementation extends the error variant type, because the deserializer will detect the content types as being incompatible and default to `null`, rather than looking at the value in the option and filtering out "new" ones.

So un-optioned variant returns cannot be extended as you desire, and optioned ones probably don't have the behavior you expect until Candid is fixed. 

My (perhaps naive) suggestion would be to enumerate all the known errors and have catch-all variant for future errors that carries a code and a text value.

I'm no Candid expert, so will defer to @chenyan-dfinity  who should definitely chime in here.

Reply:
This is true at the moment, but we are updating the spec to allow adding/removing tags in `opt variant` to be backward compatible. The implementation should rollout in a week or two. So as long as the return type is `opt variant`, instead of `variant`, it should be backward compatible very soon.

Reply:
I agree with @crusso, it's best to add a catch-all variant, and once the candid implementation is fixed, we can add more concrete error variants.

Reply:
The current candid has each of these variants as `opt` (see endpoints below). Is that not sufficient for backward compatibility? I agree though, a catch-all variant would be more easily understood than an optional variant.

Reply:
Correct.